### ISSUE TITLE
systemd fails to start in centos:7 image

### ISSUE THREAD

**Issue by ktdreyer** [containers/libpod#6183]
/kind bug

**Description**
I can run systemd inside the `centos:8` container, but I cannot run it within the `centos:7` container.

**Steps to reproduce the issue:**

1. Install podman on Fedora (I'm using Fedora 31)
2. Start systemd inside the CentOS 7 image:
```
podman run -it --privileged centos:7 /usr/sbin/init
```

**Describe the results you received:**
```
Failed to mount cgroup at /sys/fs/cgroup/systemd: Operation not permitted
[!!!!!!] Failed to mount API filesystems, freezing.
```

**Describe the results you expected:**
systemd starts in `centos:7`, the same as it does in `centos:8`

**Additional information you deem important (e.g. issue happens only occasionally):**
This happens because I'm trying to test an Ansible role with molecule, and I need to test this Ansible code on both CentOS 7 and CentOS 8.

**Output of `podman version`:**

```
Version:            1.9.1
RemoteAPI Version:  1
Go Version:         go1.13.10
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
podman info --debug
debug:
  compiler: gc
  gitCommit: ""
  goVersion: go1.13.10
  podmanVersion: 1.9.1
host:
  arch: amd64
  buildahVersion: 1.14.8
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.15-1.fc31.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.15, commit: 4152e6044da92e0c5f246e5adf14c85f41443759'
  cpus: 8
  distribution:
    distribution: fedora
    version: "31"
  eventLogger: file
  hostname: <omitted>
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.6.11-200.fc31.x86_64
  memFree: 9697738752
  memTotal: 16654970880
  ociRuntime:
    name: crun
    package: crun-0.13-2.fc31.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.0.1-1.fc31.x86_64
    version: |-
      slirp4netns version 1.0.1
      commit: 6a7b16babc95b6a3056b33fb45b74a6f62262dd4
      libslirp: 4.1.0
  swapFree: 8413769728
  swapTotal: 8413769728
  uptime: 69h 26m 16.75s (Approximately 2.88 days)
registries: <omitted>
  search:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  configFile: /home/kdreyer/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.0.0-1.fc31.x86_64
      Version: |-
        fusermount3 version: 3.6.2
        fuse-overlayfs: version 1.0.0
        FUSE library version 3.6.2
        using FUSE kernel interface version 7.29
  graphRoot: /home/kdreyer/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 16
  runRoot: /run/user/1000
  volumePath: /home/kdreyer/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.9.1-1.fc31
```

**Comment by mheon** [comment]
This is a known issue, though I don't know if it made it to the troubleshooting guide.

Basically, the systemd in the Centos 7 image is rather old (dates back ~6 years to the release of Cent 7, though it's seen backports since) - too old to support Cgroups v2. Fedora 31 uses Cgroups v2 by default, and in systemd containers, Podman forwards a subset of the host's cgroups into the container (so systemd within the container can set resource limits, among other things). Combine these together, and we're giving the old systemd too-new cgroups from the host, causing it to fail to start.

A few potential remediations: You can use a more recent Centos image (Centos 8 should have support for Cgroups v2). Alternatively, you could turn off Cgroups v2 on the host (which does have some disadvantages - no more resource limits for rootless containers).

We've looked into resolving this on the Podman side (there are some potential ways to mimic v1 cgroups within a container, even on a v2 host), but they're prohibitively complicated to implement.

Hope this helps!

**Comment by ktdreyer** [comment]
Thank you for explaining this. It would be great to add this information to the troubleshooting guide.

Should I expect this to work on a Cgroups v1 host (like CentOS 7)?

**Comment by mheon** [comment]
Yes, I'd expect it to work fine on a Cent 7 host.

**Comment by ktdreyer** [comment]
Thanks for the detailed answers. I'm closing this ticket now.

**Comment by rhatdan** [comment]
@ktdreyer It is covered in the troublshooting guide.
dependabot/go_modules/github.com/rootless-containers/rootlesskit-0.9.5

**Comment by ktdreyer** [comment]
Is there a web URL where this is documented? I Googled rootlesskit like a noob but I cannot find this information in that project's [README](https://github.com/rootless-containers/rootlesskit).

Before I opened this ticket, when I was searching Google about this problem, I found docs like [rootless.md](https://github.com/containers/libpod/blob/master/rootless.md), but it was not clear to me if anything there fit my problem description.
